### PR TITLE
Merge argparse2cwl + gxargparse

### PR DESCRIPTION
**Pull request by anton-khodak**
https://github.com/common-workflow-language/gxargparse/issues/18

### CODE REVIEW

**Comment by erasche** [comment]
Sorry for delayed response, will look at this soon. Thanks for upstreaming y'all's stuff.

**Comment by erasche** [comment]
Ok, looks good.

**Comment by erasche** [comment]
By "looks good", I mean, I'm willing to merge this to work with y'all:

- I'm not as active in this project as I could be and I have no interest in impeding process, hence the merge.
- However, no recent commits != dead project / galaxy parts aren't being used. The galaxy parts are definitely being used and should be treated at the same level as the cwl stuff. 

There are some issues I'm correcting post-merge or will need you to correct:

- This project has historically supported 2.7, but the tests that were added only run on 3.3+, due to the use of Mock. 
    - 90% of the uses could have been avoided by using the [built-in feature](https://docs.python.org/2.7/library/argparse.html#parsing-arguments) to pass an argv to argparse
    - The others could stand to just patch objects manually rather than using the newer library. I took a stab at fixing that and got a lot of errors from cwl side, so I'm not looking at it further. If you are still working on this, it would be good if you fixed this and added `nosetests` (or similar) to the `.travis.yml`
- The readme was changed significantly to highlight CWL rather than a single run through saying "here's how to do this in galaxy XML, and here's the corresponding CWL commands". I have reverted this in the README, and highlight both tooling formats simultaneously.
- Things like `self.parse_args_galaxy_nouse` vs `self.parse_args_cwl` make it seem like galaxy is a second class citizen when it's the sole reason for this project existing. This is easily corrected, but it gives me a negative impression of the author's attitude to this project.

The CWL are really quite separate from galaxy stuff, and that's a bit unfortunate, but oh well. Maybe someday we can get more generic code paths.

I am renaming this project argparse2tool, pursuant to https://github.com/common-workflow-language/gxargparse/issues/1

**Comment by mr-c** [comment]
Hello @erasche 

Thank you very much for the use of your codebase and for the merge!

My apologies for not fully preparing our branch before requesting a merge back in; I should have done a thorough review.

I've updated our documentation to point to the new name; thank you for changing that.

We hope to eventually have a generic CWL serialization library and move away from the jinja templates -- hopefully that will enable cleaner and more shareable code paths.

Again, my apologies for the awkward merge.

**Comment by erasche** [comment]
No worries Michael. Again, happy to accomodate y'all however I can. I'm
glad to see cwl support in here. This will significantly expand the utility.

Den tir. 15. nov. 2016, 08.03 skrev Michael R. Crusoe <
notifications@github.com>:

**Comment by erasche** [comment]
And my apologies for grouchiness. Unrelated events sometimes get rolled up
into too emotional replies. I usually do my best to divorce emotions from
my code, but sometimes things come off overly negative.

Den tir. 15. nov. 2016, 08.04 skrev Eric Rasche <rasche.eric@gmail.com>:

File: test/test_arg2cwl.py [hunk 1]
```diff
@@ -218,6 +226,56 @@ def test_outputs(self):
...
+        actions = list(chain(self._strip_help_version_actions(mum._actions),
+                           self._strip_help_version_actions(dad._actions)))
+        arguments = [arg.dest for arg in actions]
+        arguments.append('kids_argument')
+        for arg in arguments:
+            self.assertIn(arg, tool.inputs.keys())
+
+
+
+    def test_prefixes(self):
```
Comment (anton-khodak) [inline test/test_arg2cwl.py]: #11 @mr-c argparse2tool supports prefix_chars

**Pull request merged**